### ISSUE TITLE
Unable to pull branch 'xxx' from origin

### ISSUE THREAD

**Issue by yakamoz423** [gitkraken/vscode-gitlens#3389]
### Description

I noticed that GitLens "pull with rebase" uses the command:
 `git pull --tags -r origin fixdev:fixdev`

But I got:
```
From http://xxx/Code
 ! [rejected]            xxx     -> xxx  (non-fast-forward)
```

Instead, I use the command:
 `git pull --tags -r`

The result is fine:
```
Successfully rebased and updated refs/heads/fixdev.
```

### GitLens Version

v15.2.0

### VS Code Version

Version: 1.91.0 (system setup)
Commit: ea1445cc7016315d0f5728f8e8b12a45dc0a7286
Date: 2024-07-01T18:52:22.949Z
Electron: 29.4.0
ElectronBuildId: 9728852
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Windows_NT x64 10.0.22631

### Git Version

git version 2.45.2.windows.1

**Comment by axosoft-ramint** [comment]
@yakamoz423 Would you mind doing the following to help us investigate:
1. Ensuring your GitLens log output level is set to 'debug' (if it's not already, you can set `  "gitlens.outputLevel": "debug",` in your settings JSON or use the command palette command `GitLens: Enable Debug Logging`).
2. Reproduce the issue and post the output from your `GitLens (Git)` and `GitLens` output channels (feel free to redact anything sensitive such as repo names if needed)

Thanks!

**Comment by axosoft-ramint** [comment]
For context, here are the steps I took so far to try to repro:

1. Make and commit some changes on the upstream of a local branch, so that local branch is now 1 commit behind.
2. Open that repo in GitLens, with the local branch checked out.
3. Use the `GitLens: Git Command Palette` command to bring up git commands
4. Choose `pull`
5. Choose `Pull with Rebase`
6. The command that is run (and shown in output) is `git pull --tags -r origin abcde:abcde`, which is expected, and succeeds without issue.

I'm probably missing something though. For examples:

- is `origin` the correct upstream for the branch in your case? Or is there a different default remote or upstream that the command uses when you don't include `origin fixdev:fixdev`?
- Is the upstream branch name different than the local?
- Are there hooks/protections that would cause this rejection to occur?

**Comment by yakamoz423** [comment]
> @yakamoz423 Would you mind doing the following to help us investigate:
> 
> 1. Ensuring your GitLens log output level is set to 'debug' (if it's not already, you can set `  "gitlens.outputLevel": "debug",` in your settings JSON or use the command palette command `GitLens: Enable Debug Logging`).
> 2. Reproduce the issue and post the output from your `GitLens (Git)` and `GitLens` output channels (feel free to redact anything sensitive such as repo names if needed)
> 
> Thanks!

This is actually unrelated to GitLens. I don't know whether this is a bug or a feature of Git, or if there's something else going wrong.

I was wondering if it would be possible to simplify GitLens's `Pull with rebase` command so that it doesn't specify the remote branch explicitly? As far as I remember, previous versions of GitLens didn't encounter this issue.

**Comment by epsilonxaos** [comment]
This error message was also shown when using only the `Pull` commands or `git pull --tags origin dev:dev ` in debug mode, currently the most current version no longer has this problem and is fulfilled correctly, I am not sure that the problem was with git since if you manually executed the command the `Pull` was carried out without any problem.

I recommend updating the extension.

**Comment by pdesmarais** [comment]
Going through the exact same issue here. Updated git to 2.45.2 (from 2.41), running GitLens 15.2.0. Also tried with prerelease version of GitLens to no avail.

**Comment by axosoft-ramint** [comment]
Can someone verify that this has been fixed on the pre-release version of GitLens? You can switch to the pre-release edition of GitLens by clicking on the "Switch to Pre-Release version of this extension" from the Extensions view.

![image](https://user-images.githubusercontent.com/641685/228608740-6071bc19-7b1a-44c8-843f-d2af44d6114b.png)

**Comment by axosoft-ramint** [comment]
Fixed in commit: 5e24602b0d4e5206f4a1402add4c37b7858a47c8

**Comment by yakamoz423** [comment]
It fixed on v2024.7.1905 (pre-release), thanks!
